app/src/components/GameRoutes.tsx

```diff
@@ -15,23 +27,82 @@ function CurrentPlayerProvider(props: {
   joinCode: string
   children: React.ReactNode
 }) {
-  const { data, loading } = useCurrentPlayerQuery({
+  const currentAuth = React.useContext(CurrentAuthContext)
+
+  const [currentPlayer, setCurrentPlayer] = React.useState<
+    CurrentPlayerQuery["players"][0] | null
```

Comment:
interesting, is this a shortcut to get a type?

Reply:
Yea, it's the types auto generated by graphql-codegen based on the Hasura schema.

In general, it's better to use gql Fragments here... but I haven;t gotten around to that.